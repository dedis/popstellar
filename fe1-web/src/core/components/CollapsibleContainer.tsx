import PropTypes from 'prop-types';
import React, { useState } from 'react';
import { StyleSheet, Text, TextStyle, TouchableOpacity, View } from 'react-native';

import { Typography } from '../styles';
import TextBlock from './TextBlock';

const CollapsibleContainerStyles = StyleSheet.create({
  headerIcon: {
    position: 'absolute',
    right: 0,
  },
});

const CollapsibleContainer = ({ title, children }: IPropTypes) => {
  const [showChildren, setShowChildren] = useState(false);

  return (
    <View>
      <TouchableOpacity onPress={() => setShowChildren(!showChildren)}>
        <View>
          <TextBlock bold text={title || ''} />
          <Text style={[Typography.base as TextStyle, CollapsibleContainerStyles.headerIcon]}>
            {showChildren ? 'v' : '<'}
          </Text>
        </View>
      </TouchableOpacity>

      {showChildren && children}
    </View>
  );
};

const propTypes = {
  title: PropTypes.string,
  children: PropTypes.node,
};
CollapsibleContainer.propTypes = propTypes;

CollapsibleContainer.defaultProps = {
  title: '',
  children: null,
};

type IPropTypes = PropTypes.InferProps<typeof propTypes>;

export default CollapsibleContainer;
