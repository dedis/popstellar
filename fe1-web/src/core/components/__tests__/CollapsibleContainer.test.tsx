import { describe } from '@jest/globals';
import { render } from '@testing-library/react-native';
import React from 'react';
import { Text } from 'react-native';

import CollapsibleContainer from '../CollapsibleContainer';

describe('CollapsibleContainer', () => {
  it('renders correctly', () => {
    const component = render(
      <CollapsibleContainer title="A collapsible container">
        <Text>a first child</Text>
        <Text>a second child</Text>
      </CollapsibleContainer>,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
