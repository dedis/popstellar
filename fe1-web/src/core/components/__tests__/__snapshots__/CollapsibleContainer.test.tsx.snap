// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsibleContainer container can be opened 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 25,
            "fontWeight": "bold",
            "marginHorizontal": 10,
            "textAlign": "center",
          }
        }
      >
        open me
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 25,
              "marginHorizontal": 10,
              "textAlign": "center",
            },
            Object {
              "position": "absolute",
              "right": 0,
            },
          ]
        }
      >
        v
      </Text>
    </View>
  </View>
  <Text>
    a first child
  </Text>
  <Text>
    a second child
  </Text>
</View>
`;

exports[`CollapsibleContainer renders correctly 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 25,
            "fontWeight": "bold",
            "marginHorizontal": 10,
            "textAlign": "center",
          }
        }
      >
        A collapsible container
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 25,
              "marginHorizontal": 10,
              "textAlign": "center",
            },
            Object {
              "position": "absolute",
              "right": 0,
            },
          ]
        }
      >
        &lt;
      </Text>
    </View>
  </View>
</View>
`;
